tool/tsh/common/proxy.go

```diff
@@ -62,13 +63,33 @@ func onProxyCommandSSH(cf *CLIConf) error {
 			return trace.Wrap(err)
 		}
 
-		targetHost, targetPort, err := net.SplitHostPort(tc.Host)
-		if err != nil {
-			targetHost = tc.Host
-			targetPort = strconv.Itoa(tc.HostPort)
+		var target string
+		switch {
+		case tc.Host != "":
+			targetHost, targetPort, err := net.SplitHostPort(tc.Host)
+			if err != nil {
+				targetHost = tc.Host
+				targetPort = strconv.Itoa(tc.HostPort)
+			}
+			targetHost = cleanTargetHost(targetHost, tc.WebProxyHost(), clt.ClusterName())
+			target = net.JoinHostPort(targetHost, targetPort)
+		case len(tc.SearchKeywords) != 0 || tc.PredicateExpression != "":
+			nodes, err := client.GetAllResources[types.Server](cf.Context, clt.AuthClient, tc.ResourceFilter(types.KindNode))
+			if err != nil {
+				return trace.Wrap(err)
+			}
+
+			if len(nodes) == 0 {
+				return trace.NotFound("no matching targets found")
+			}
+
+			if len(nodes) > 1 {
+				return trace.BadParameter("found multiple matching targets")
```

Comment:
Nit: Include a list of matching ones?
```suggestion
				return trace.BadParameter("found multiple SSH nodes matching proxy templates query expression: %v", xxx)
```